AEPIdentity/Tests/IdentityStateTests.swift

```diff
@@ -639,6 +639,29 @@ class IdentityStateTests: XCTestCase {
         XCTAssertTrue(mockHitQueue.queuedHits.isEmpty) // hit should NOT be queued in the hit queue
     }
 
+    /// Tests that a hit is queued when forceSynce
+    func testSyncIdentifiers_whenForceSync_shouldQueueHit() {
+        // setup
+        let configSharedState = [IdentityConstants.Configuration.EXPERIENCE_CLOUD_ORGID: "test-org",
+                                 IdentityConstants.Configuration.EXPERIENCE_CLOUD_SERVER: "test-server",
+                                 IdentityConstants.Configuration.GLOBAL_CONFIG_PRIVACY: PrivacyStatus.optedIn.rawValue] as [String: Any]
+        var props = IdentityProperties()
+        props.ecid = ECID() // visitor ID is null initially and set for the first time in
+        // shouldSync(). Mimic a second call to shouldSync by setting the ecid
+        props.lastSync = Date() // set last sync to now
+        state = IdentityState(identityProperties: props, hitQueue: MockHitQueue(processor: MockHitProcessor()), pushIdManager: mockPushIdManager)
+        state.lastValidConfig = configSharedState
+
+        // test
+        let data = [IdentityConstants.EventDataKeys.IS_SYNC_EVENT: true] as [String: Any]
+        _ = state.syncIdentifiers(event: Event(name: "ID Sync Test Event", type: EventType.identity, source: EventSource.requestIdentity, data: data), forceSync: true)
```

Comment:
You are setting both the event data `forcesync` flag and the function parameter `forceSync` to true which makes it hard to determine which flag is causing the forced sync. Instead, can you split this into two tests, one which sets the event data `forcesync` to true and the function parameter `forceSync` to false and another which does the opposite?

Reply:
@kevinlind I am setting IS_SYNC_EVENT eventdata flag and not FORCE_SYNC flag, but good catch I added one for FORCE_SYNC eventdata.